src/psyclone/tests/psyGen_test.py

```diff
@@ -2621,8 +2631,465 @@ def test_binaryoperation_can_be_printed():
+
+    assert isinstance(sym_table.lookup("var1"), Symbol)
+    assert sym_table.lookup("var1").name == "var1"
+    assert isinstance(sym_table.lookup("var2"), Symbol)
+    assert sym_table.lookup("var2").name == "var2"
+    assert isinstance(sym_table.lookup("var3"), Symbol)
+    assert sym_table.lookup("var3").name == "var3"
+
+    with pytest.raises(KeyError) as error:
+        sym_table.lookup("notdeclared")
+
+
+def test_symboltable_view(capsys):
+    '''Test the view method of the SymbolTable class, it should print to
+    standard out a representation of the full SymbolTable'''
+    sym_table = SymbolTable()
+    sym_table.declare("var1", "real", [], "read_arg")
+    sym_table.declare("var2", "integer", [], "read_arg")
+    sym_table.view()
+    output, _ = capsys.readouterr()
+    assert "Symbol Table:\n" in output
+    assert "var1" in output
+    assert "var2" in output
+
+
+def test_symboltable_can_be_printed():
+    '''Test that a SymbolTable instance can always be printed (i.e. is
+    initialised fully)'''
+    sym_table = SymbolTable()
+    sym_table.declare("var1", "real", [], "read_arg")
+    sym_table.declare("var2", "integer", [], "read_arg")
+    assert "Symbol Table:\n" in str(sym_table)
+    assert "var1" in str(sym_table)
+    assert "var2" in str(sym_table)
+
+
 # Test Fparser2ASTProcessor
 
+def test_fparser2astprocessor_generate_schedule_empty_subroutine():
+    ''' Tests the fparser2AST generate_schedule method with an empty
+    subroutine.
+    '''
+    ast1 = fpapi.parse(FAKE_KERNEL_METADATA, ignore_comments=True)
+    metadata = DynKernMetadata(ast1)
+    my_kern = DynKern()
+    my_kern.load_meta(metadata)
+    ast2 = my_kern.ast
+    processor = Fparser2ASTProcessor()
+
+    # Test properly formed but empty kernel module
+    schedule = processor.generate_schedule("dummy_code", ast2)
+    assert isinstance(schedule, KernelSchedule)
+
+    # Test that we get an error for a nonexistant subroutine name
+    with pytest.raises(InternalError) as error:
+        schedule = processor.generate_schedule("nonexistent_code", ast2)
+    assert "Unexpected kernel AST. Could not find " \
+           "subroutine: nonexistent_code" in str(error.value)
+
+    # Test corrupting ast by deleting subroutine
+    del ast2.content[0].content[2]
+    with pytest.raises(InternalError) as error:
+        schedule = processor.generate_schedule("dummy_code", ast2)
+    assert "Unexpected kernel AST. Could not find " \
+           "subroutine: dummy_code" in str(error.value)
+
+    # Test corrupting ast by deleting specification part
+    del ast2.content[0].content[1]
+    with pytest.raises(InternalError) as error:
+        schedule = processor.generate_schedule("dummy_code", ast2)
+    assert "Unexpected kernel AST. Could not find specification part."
+
+
+def test_fparser2astprocessor_generate_schedule_two_modules():
+    ''' Tests the fparser2AST generate_schedule method with an simple
+    subroutine.
```

Comment:
Please update this docstring to say that we're testing that we reject files containing more than one module definition.

Reply:
Done